### PR TITLE
fix/set-person-error

### PR DESCRIPTION
**Pull request by gavin-brown**
This fixes an error that could occur sometimes and break sites if not handled.

Easiest way to test:
- Inside this project, build using `yarn build`
- Copy the built `lib` folder into the `studio-caliper-js` folder within the `node_modules` folder of the project you're testing with

However, I'm not sure how to reproduce the original error. I think I got the error by having existing (and different) data in my localhost:3000 domain for studioKit:caliperService:person

### CODE REVIEW

**Comment by gavin-brown** [comment]
@builtbywill This change resolved my Caliper error. Is this all we would need here?

**Comment by builtbywill** [comment]
@gavin-brown I guess if we are making updates, we could do some dependency updates, probably in caliper-js, too?

**Comment by gavin-brown** [comment]
@builtbywill dependabot will be very happy 😅

**Comment by gavin-brown** [comment]
@purdue-tlt/full-timers Ready for review

**Comment by gavin-brown** [comment]
(That's a joke. This is my own merge request)

File: src/browser-caliper-service.test.js [hunk 1]
```diff
@@ -15,22 +15,12 @@ chai.should()
...
-	})
-}
-
-function restoreRavenIsSetup() {
-	Raven.isSetup.restore()
-}
-
-function stubRavenCaptureException() {
-	return sinon.stub(Raven, 'captureException').callsFake(function() {})
+function stubSentryCaptureException() {
```
Comment (lily514) [inline src/browser-caliper-service.test.js:18]: @gavin-brown is there a sentry project for caliper, or will it just log these entries to whatever project is using this library?

Comment (builtbywill) [reply src/browser-caliper-service.test.js:18]: @lily514 yeah, it will use whatever project it is running inside of

File: src/caliper-service.js [hunk 2]
```diff
@@ -602,11 +602,9 @@ function CaliperService(options, asyncLoadCallback) {
 		if (!isNil(extensions) && !isPlainObject(extensions)) {
 			throw new Error('Cannot set Person: `extensions` must be a plain object.')
 		}
-		// allow person to be updated, but not changed if session is in progress
+		// If a session is in-progress, end the session when changing person
```
Comment (lily514) [inline src/caliper-service.js:605]: @gavin-brown and this section is essentially the fix, and everything else is just dependency updates right? Trying to wrap my head around this.

Comment (gavin-brown) [reply src/caliper-service.js:605]: @lily514 Yep, exactly

**Review by lily514 — APPROVED** [review]
Sweet. Thanks Gavin! Sorry we forgot about this for so long. :denver:

File: src/caliper-service.js (hunk 2 above)
Comment (lily514) [reply src/caliper-service.js:605]: The fix makes sense though, the error before was like, "you should have ended it first", so why not just end it if you get here. :thumps:

Also, I don't think the git hooks are showing approvals. :fry:

**Pull request merged**